content/en/blog/2024/prom-and-otel/index.md

```diff
@@ -0,0 +1,671 @@
+5. The Collectors scrape their assigned targets
+
+This means that the OTel collectors -- not a Prometheus scraper -- collect the
+metrics.
+
+A **Target** is an endpoint that supplies Metrics for Prometheus to store. A
+**Scrape** is the action of collecting Metrics through an HTTP request from a
+targeted instance, parsing the response, and ingesting the collected samples to
+storage.
+
+```mermaid
+sequenceDiagram
+  participant Target Allocator
+  participant Metrics Targets
+  participant OTel Collectors
+  Target Allocator ->>Metrics Targets: 1. Discover Metrics targets
+  Target Allocator ->>OTel Collectors: 2. Discover available Collectors
+  Target Allocator ->>Target Allocator: 3. Assign Metrics targets
+  OTel Collectors ->>Target Allocator: 4. Query TA for Metrics endpoints scrape
+  OTel Collectors ->>Metrics Targets: 5. Scrape Metrics target
+```
+
+### Discovery of Prometheus custom resources
+
+The Target Allocator’s second job is to provide the discovery of Prometheus
+Operator CRs, namely the
+[ServiceMonitor and PodMonitor](https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator#target-allocator).
+
+In the past, all Prometheus scrape configurations had to be done via the
+Prometheus Receiver. When the Target Allocator’s service discovery feature is
+enabled, the TA simplifies the configuration of the Prometheus receiver, by
+creating scrape configurations in the Prometheus receiver from the PodMonitor
+and ServiceMonitor instances deployed in your cluster.
+
+```mermaid
+flowchart RL
+  pm(PodMonitor)
+  sm(ServiceMonitor)
+  ta(Target Allocator)
+  oc1(OTel Collector)
+  oc2(OTel Collector)
+  oc3(OTel Collector)
+  ta --> pm
+  ta --> sm
+  oc1 --> ta
+  oc2 --> ta
+  oc3 --> ta
+  sm ~~~|"1. Discover Prometheus Operator CRs"| sm
+  ta ~~~|"2. Add job to TA scrape configuration"| ta
+  oc3 ~~~|"3. Add job to OTel Collector scrape configuration"| oc3
+```
+
+Even though Prometheus is not required to be installed in your Kubernetes
+cluster to use the Target Allocator for Prometheus CR discovery, the TA does
+require that the ServiceMonitor and PodMonitor be installed. These CRs are
+bundled with Prometheus Operator; however, they can be installed standalone as
+well. The easiest way to do this is to grab a copy of the individual
+[PodMonitor YAML](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/charts/crds/crds/crd-podmonitors.yaml)
+and
+[ServiceMonitor YAML](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/charts/crds/crds/crd-servicemonitors.yaml)
+custom resource definitions (CRDs).
+
+OTel supports the PodMonitor and ServiceMonitor Prometheus resources because
+these are widely-used in Kubernetes infrastructure monitoring. As a result, the
+OTel Operator developers wanted to make it easy to add them to the OTel
+ecosystem.
+
+Note that the PodMonitor and ServiceMonitor are not useful for cluster-wide
```

Comment:
"cluster-wide" feels misleading.  I would describe the limitation as "PodMonitor and ServiceMonitor are limited to collecting metrics from pods, and are unable to scrape other endpoints, such as the kubelet."